test/unit/api/api-record-log-events.test.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper.js')
+const API = require('../../../api.js')
+const { SUPPORTABILITY, LOGGING } = require('../../../lib/metrics/names')
+const API_METRIC = SUPPORTABILITY.API + '/recordLogEvent'
+
+tap.test('Agent API - recordCustomEvent', (t) => {
+  t.autoend()
+
+  let agent = null
+  let api = null
+  const message = 'just logging a log in the logger'
+
+  t.beforeEach(() => {
+    agent = helper.loadMockedAgent()
+    api = new API(agent)
+  })
+
+  t.afterEach(() => {
+    helper.unloadAgent(agent)
+    agent = null
+    api = null
+  })
+
+  t.test('can handle a singular log message', (t) => {
```

Comment:
I'd like to see a test where we set message, level, timestamp, other random key/value pairs.  At the very least a test where timestamp exists and does not default to Date.now()

Reply:
Okay, added more details to the test to check for other params and a non-standard date.